subcommand/tls-init/commant_test.go

```diff
@@ -0,0 +1,582 @@
+package tls_init
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/mitchellh/cli"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func TestRun_FlagValidation(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		flags  []string
+		expErr string
+	}{
+		{
+			flags:  nil,
+			expErr: "-name-prefix must be set",
+		},
+		{
+			flags:  []string{"-name-prefix", "consul", "-ca", "foo"},
+			expErr: "either both -ca and -key or neither must be set",
+		},
+		{
+			flags:  []string{"-name-prefix", "consul", "-key", "/foo"},
+			expErr: "either both -ca and -key or neither must be set",
+		},
+		{
+			flags:  []string{"-name-prefix", "consul", "-log-level", "invalid"},
+			expErr: "unknown log level: invalid",
+		},
+		{
+			flags:  []string{"-name-prefix", "consul", "-days", "-3"},
+			expErr: "-days must be a positive integer",
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.expErr, func(tt *testing.T) {
+			ui := cli.NewMockUi()
+			cmd := Command{UI: ui}
+			exitCode := cmd.Run(c.flags)
+			require.Equal(tt, 1, exitCode, ui.ErrorWriter.String())
+			require.Contains(tt, ui.ErrorWriter.String(), c.expErr)
+		})
+	}
+}
+
+func TestRun_CreatesServerCertificatesWithExistingCertsAsFiles(t *testing.T) {
+	ui := cli.NewMockUi()
+	cmd := Command{UI: ui}
+	k8s := fake.NewSimpleClientset()
+	cmd.clientset = k8s
+
+	ca, err := ioutil.TempFile("", "")
+	require.NoError(t, err)
+	err = ioutil.WriteFile(ca.Name(), []byte(caCert), 0644)
+	require.NoError(t, err)
+
+	key, err := ioutil.TempFile("", "")
+	require.NoError(t, err)
+	err = ioutil.WriteFile(key.Name(), []byte(caKey), 0644)
+	require.NoError(t, err)
+
+	flags := []string{"-name-prefix", "consul", "-ca", ca.Name(), "-key", key.Name()}
+
+	exitCode := cmd.Run(flags)
+	require.Equal(t, 0, exitCode)
+
+	serverCertSecret, err := k8s.CoreV1().Secrets("default").Get(context.Background(), "consul-server-cert", metav1.GetOptions{})
+	require.NoError(t, err)
+	serverCert := serverCertSecret.Data[corev1.TLSCertKey]
+	serverKey := serverCertSecret.Data[corev1.TLSPrivateKeyKey]
+
+	certBlock, _ := pem.Decode([]byte(serverCert))
+	certificate, err := x509.ParseCertificate(certBlock.Bytes)
+	require.NoError(t, err)
+	require.False(t, certificate.IsCA)
+	require.Equal(t, []string{"server.dc1.consul", "localhost"}, certificate.DNSNames)
```

Comment:
should we check IPs as well?

Reply:
cant think of a single good reason not to 😉

Reply:
done :)